RecoEgamma/ElectronIdentification/plugins/ElectronIDValueMapProducer.cc

```diff
@@ -166,30 +126,21 @@ void ElectronIDValueMapProducer::produce(edm::Event& iEvent, const edm::EventSet
   writeValueMap(iEvent, src, eleFull5x5E2x5, eleFull5x5E2x5_);   
   writeValueMap(iEvent, src, eleFull5x5E5x5, eleFull5x5E5x5_);  
   writeValueMap(iEvent, src, eleFull5x5Circularity, eleFull5x5Circularity_);  
-  
-  delete lazyToolnoZS;
-}
-
-void ElectronIDValueMapProducer::writeValueMap(edm::Event &iEvent,
-					     const edm::Handle<edm::View<reco::GsfElectron> > & handle,
-					     const std::vector<float> & values,
-					     const std::string    & label) const 
-{
-  using namespace edm; 
-  using namespace std;
-  auto valMap = std::make_unique<ValueMap<float>>();
-  edm::ValueMap<float>::Filler filler(*valMap);
-  filler.insert(handle, values.begin(), values.end());
-  filler.fill();
-  iEvent.put(std::move(valMap), label);
 }
 
 void ElectronIDValueMapProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
-  //The following says we do not know what parameters are allowed so do no validation
-  // Please change this to state exactly what you do use, even if it is no parameters
+  // electronIDValueMapProducer
   edm::ParameterSetDescription desc;
-  desc.setUnknown();
-  descriptions.addDefault(desc);
+  desc.add<edm::InputTag>("src", edm::InputTag("gedGsfElectrons"));
+  desc.add<edm::InputTag>("srcMiniAOD", edm::InputTag("slimmedElectrons","","@skipCurrentProcess"));
+  desc.add<edm::InputTag>("esReducedRecHitCollection", edm::InputTag("reducedEcalRecHitsES"));
+  desc.add<edm::InputTag>("eeReducedRecHitCollection", edm::InputTag("reducedEcalRecHitsEE"));
+  desc.add<edm::InputTag>("ebReducedRecHitCollectionMiniAOD", edm::InputTag("reducedEgamma","reducedEBRecHits"));
+  desc.add<edm::InputTag>("eeReducedRecHitCollectionMiniAOD", edm::InputTag("reducedEgamma","reducedEERecHits"));
+  desc.add<edm::InputTag>("esReducedRecHitCollectionMiniAOD", edm::InputTag("reducedEgamma","reducedESRecHits"));
+  desc.add<edm::InputTag>("ebReducedRecHitCollection", edm::InputTag("reducedEcalRecHitsEB"));
```

Comment:
it would be nice to have this above and keep either the same order {eb, ee, es} for aod first then the same for miniAOD or fill {eb aod, eb mini} and so on in order rather than a mix like now.

Reply:
the previous comment here https://github.com/cms-sw/cmssw/pull/24312/files#r212407159 was not addressed